client/my-sites/pages/page-list.jsx

```diff
@@ -229,8 +227,14 @@ var Pages = React.createClass( {
 			const site = this.props.sites.getSelectedSite();
 			const status = this.props.status || 'published';
 
-			if ( status === 'published' && get( site, 'options.show_on_front' ) === 'posts' ) {
-				rows.push( <BlogPostsPage key="blog-posts-page" /> );
+			if ( site && status === 'published' ) {
+				rows.push(
+					<BlogPostsPage
+						key="blog-posts-page"
+						homePageType={ site.options.show_on_front }
```

Comment:
`lodash.get` was being used before because if viewing All Sites, `site` is null. In the change, the check for site being set is done before this code is executed.